ui/app/AppLayouts/Chat/components/PrivateChatPopup.qml

```diff
@@ -67,107 +75,112 @@ ModalPopup {
         pubKey = "";
         ensUsername.text = "";
         chatKey.forceActiveFocus(Qt.MouseFocusReason)
-        noContactsRect.visible = !profileModel.contacts.list.hasAddedContacts()
+        existingContacts.visible = profileModel.contacts.list.hasAddedContacts()
+        noContactsRect.visible = !existingContacts.visible
     }
 
     Input {
         id: chatKey
         //% "Enter ENS username or chat key"
         placeholderText: qsTrId("enter-contact-code")
-        Keys.onEnterPressed: doJoin()
-        Keys.onReturnPressed: doJoin()
-        validationError: popup.validationError
+        Keys.onEnterPressed: validateAndJoin(popup.pubKey, chatKey.text)
+        Keys.onReturnPressed: validateAndJoin(popup.pubKey, chatKey.text)
         Keys.onReleased: {
             onKeyReleased();
         }
+        textField.anchors.rightMargin: clearBtn.width + Style.current.padding + 2
 
         Connections {
             target: chatsModel
             onEnsWasResolved: {
                 if(chatKey.text == ""){
-                    ensUsername.text == "";
+                    ensUsername.text = "";
                     pubKey = "";
                 } else if(resolvedPubKey == ""){
-                    //% "User not found"
-                    ensUsername.text = qsTrId("user-not-found");
+                    ensUsername.text = "";
                     pubKey = "";
+                    searchResults.showProfileNotFoundMessage = true
                 } else {
                     if (profileModel.profile.pubKey === resolvedPubKey) {
-                        validationError = qsTr("Can't chat with yourself");
+                        popup.validationError = qsTr("Can't chat with yourself");
                     } else {
-                        ensUsername.text = chatsModel.formatENSUsername(chatKey.text) + " • " + Utils.compactAddress(resolvedPubKey, 4)
+                        searchResults.username = chatsModel.formatENSUsername(chatKey.text)
+                        let userAlias = chatsModel.generateAlias(resolvedPubKey)
+                        userAlias = userAlias.length > 20 ? userAlias.substring(0, 19) + "..." : userAlias
+                        searchResults.userAlias =  userAlias + " • " + Utils.compactAddress(resolvedPubKey, 4)
                         pubKey = resolvedPubKey;
                     }
+                    searchResults.showProfileNotFoundMessage = false
                 }
-                loading = false;
+                searchResults.loading = false;
+                noContactsRect.visible = pubKey === ""  && ensUsername.text === "" && !profileModel.contacts.list.hasAddedContacts() && !profileNotFoundMessage.visible
+            }
+        }
+
+        StatusIconButton {
+            id: clearBtn
+            icon.name: "close-icon"
+            type: "secondary"
+            visible: chatKey.text !== ""
+            hoveredIconColor: Style.current.black
```

Comment:
Not sure if this looks good in dark mode

Reply:
Good point, will check this one!